KerbalVR_Mod/KerbalVR/InternalModules/KerbalVR_Lever.cs

```diff
@@ -0,0 +1,308 @@
+		/// </summary>
+		[KSPField]
+		public string rotationTransformName;
+
+		/// <summary>
+		/// Axis of rotation of the rotation transform
+		/// </summary>
+		[KSPField]
+		public Vector3 rotationAxis = Vector3.right;
+
+		/// <summary>
+		/// Range of rotation of the rotation transform<br/>
+		/// <see cref="Vector2.x"/> is the lower limit<br/>
+		/// <see cref="Vector2.y"/> is the upper limit<br/>
+		/// </summary>
+		[KSPField]
+		public Vector2 rotationRange = Vector3.up;
+
+		/// <summary>
+		/// Does the lever need to be pulled out when moving
+		/// </summary>
+		[KSPField]
+		public bool pullable = false;
+
+		/// <summary>
+		/// Name of the pull transform of the pull transform
+		/// </summary>
+		[KSPField]
+		public string pullTransformName;
+
+		/// <summary>
+		/// Direction the pull transform will be pulled to
+		/// </summary>
+		[KSPField]
+		public Vector3 pullDirection = Vector3.forward;
+
+		/// <summary>
+		/// Range of the movement of the pull transform<br/>
+		/// <see cref="Vector2.x"/> is the normal position<br/>
+		/// <see cref="Vector2.y"/> is the pulled position<br/>
+		/// </summary>
+		[KSPField]
+		public Vector2 pullRange = Vector3.up;
+
+		/// <summary>
+		/// The time it takes for the lever transform to move in / out
+		/// </summary>
+		[KSPField]
+		public float pullDuration = 0.1f;
+
+		/// <summary>
+		/// Number of steps of the lever
+		/// </summary>
+		[KSPField]
+		public int stepCount = 2;
+
+		/// <summary>
+		/// Sound that will be played when step changes
+		/// </summary>
+		[KSPField]
+		public string stepSound = string.Empty;
+
+		/// <summary>
+		/// What this lever control
+		/// </summary>
+		[KSPField]
+		public string handler = string.Empty;
+
+		#endregion
+
+		[SerializeField] internal AudioSource m_audioSource;
+		internal IVALever ivaLever;
+		internal VRLeverInteractionListener interactionListener;
+		internal Transform rotationTransform;
+		internal Transform pullTransform;
+
+		public int CurrentStep => interactionListener.CurrentStep;
+
+#if PROP_GIZMOS
+		GameObject topGizmo;
+		GameObject rotationGizmo;
+		GameObject pullGizmo;
+#endif
+
+		public override void OnLoad(ConfigNode node)
+		{
+			base.OnLoad(node);
+
+			if (HighLogic.LoadedScene == GameScenes.LOADING)
+			{
+				if (stepSound != string.Empty)
+				{
+					AudioClip audioClip = GameDatabase.Instance.GetAudioClip(stepSound);
+
+					if (audioClip != null)
+					{
+						m_audioSource = internalProp.gameObject.AddComponent<AudioSource>();
+						m_audioSource.clip = audioClip;
+						m_audioSource.Stop();
+						m_audioSource.volume = GameSettings.SHIP_VOLUME;
+						m_audioSource.minDistance = 2;
+						m_audioSource.maxDistance = 10;
+						m_audioSource.panStereo = 0;
+						m_audioSource.playOnAwake = false;
+					}
+				}
+			}
+		}
+
+		public override void OnAwake()
+		{
+			base.OnAwake();
+
+			rotationTransform = internalProp.FindModelTransform(rotationTransformName);
```

Comment:
you might want to use the InternalModuleExtensions.FindTransform method instead here, so that it can support relative paths